### PR TITLE
Use for_each instead of count

### PR DESCRIPTION
**Pull request by jsf9k**
## 🗣 Description

In this pull request I:
* Use `for_each` instead of `count`.  This way we can add/remove/modify subnets without having to recreate them all.
* Return actual resources as outputs instead of just IDs.  Terraform allows you to do this now, and I think it makes a lot more sense.  The user can grab the ID, name, or any other attribute he or she needs instead of using a data resource.

## 💭 Motivation and Context

Having to destroy all resources generated by `count` just to add a new one is painful.  Let's stop doing that.

## 🧪 Testing

I deployed the example code to make sure everything is working.

## 🚥 Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (causes existing functionality to change)

## ✅ Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by felddy — APPROVED** [review]
Nifty.  Makes much more sense to do things this way.  HCL⬆️

File: examples/basic_usage/outputs.tf [hunk 1]
```diff
@@ -1,14 +1,14 @@
-output "vpc_id" {
-  value       = aws_vpc.the_vpc.id
-  description = "The VPC ID"
+output "vpc" {
+  value       = aws_vpc.the_vpc
+  description = "The VPC"
```
Comment (mcdonnnj) [inline examples/basic_usage/outputs.tf]: I think the descriptions in this file should also end in periods.

**Review by mcdonnnj — CHANGES_REQUESTED** [review]
Looking good but please see my nitpick about periods.

File: examples/basic_usage/outputs.tf (hunk 1 above)
Comment (jsf9k) [reply examples/basic_usage/outputs.tf]: See commit cc28900.

**Review by mcdonnnj — APPROVED** [review]
Thanks for fixing that! LGTM ✔

**Pull request merged**